src/NexusMods.DataModel/Synchronizer/SynchronizerService.cs

```diff
@@ -114,44 +114,92 @@ public IObservable<LoadoutWithTxId> LastAppliedRevisionFor(GameInstallation game
-            (g, l) => g.Value || l.Value);
-        
-        var lastApplied = LastAppliedRevisionFor(loadout.InstallationInstance);
-        
-        var revisions = Loadout.RevisionsWithChildUpdates(_conn, loadoutId);
-        
-        return Observable.CombineLatest(isBusy, lastApplied, revisions, (busy, last, rev) =>
-        {
-            var currentDb = _conn.Db;
-            if (busy)
-                return LoadoutSynchronizerState.Pending;
-            
-            // Last DB revision is the same in the applied loadout
-            if (last.Id == rev.LoadoutId && currentDb.BasisTxId == last.Tx)
-                return LoadoutSynchronizerState.Current;
-            
-            if (last.Id != loadoutId)
-                return LoadoutSynchronizerState.OtherLoadoutSynced;
-            
-            _logger.LogInformation("Checking for changes in loadout {LoadoutId}", loadoutId);
-            var diffTree = GetApplyDiffTree(loadoutId);
-            var diffFound = diffTree.GetAllDescendentFiles().Any(f => f.Item.Value.ChangeType != FileChangeType.None);
-            _logger.LogInformation("Changes found in loadout {LoadoutId}: {DiffFound}", loadoutId, diffFound);
-            if (diffFound)
-                return LoadoutSynchronizerState.NeedsSync;
-            
-
-            return LoadoutSynchronizerState.Current;
-        });
-            
+
+        var isBusy = loadoutState.ObservableForProperty(l => l.Busy, skipInitial: false)
+            .Select(e => e.Value);
+
+        var lastApplied = LastAppliedRevisionFor(loadout.InstallationInstance)
+            .Where(last => last != default(LoadoutWithTxId));
+
+        var revisions = Loadout.RevisionsWithChildUpdates(_conn, loadoutId)
+            // Use DB transaction, since child updates are not part of the loadout
+            .Select(rev => (loadout:rev, revDbTx:_conn.Db.BasisTxId));
+
+        var statusObservable = Observable.CombineLatest(isBusy,
+                lastApplied,
+                revisions,
+                (busy, last, rev) => (busy, last, rev.loadout, rev.revDbTx)
+            )
+            .DistinctUntilChanged()
+            .SelectMany(
+                async tuple =>
+                {
+                    var (busy, last, rev, revDbTx) = tuple;
+                    // if the loadout is not found, it means it was deleted
+                    if (!rev.IsValid())
+                        return LoadoutSynchronizerState.OtherLoadoutSynced;
+                    
+                    if (busy)
+                        return LoadoutSynchronizerState.Pending;
+
+                    // Last DB revision is the same in the applied loadout
+                    if (last.Id == rev.LoadoutId && revDbTx == last.Tx)
+                        return LoadoutSynchronizerState.Current;
+
+                    if (last.Id != loadoutId)
+                        return LoadoutSynchronizerState.OtherLoadoutSynced;
+
+                    var diffFound = await Task.Run(() =>
```

Comment:
Might be worth making a small note here.
We don't want anyone else touching this code to remove the `Task.Run`.

Technically the main thread isn't waiting for this directly, but a change to the loadout revision (e.g. disable/enable a mod) will fire a MMDB transaction which runs this via observable callback. The UI element whose receiving the result of this callback will the in turn end up waiting for the result.

So without `Task.Run`, the Main UI isn't actually stalled whatsoever, ***however*** the radio buttons which toggle a mod's enabled state will wait for completion, making the UI feel as responsive as it should be. Therefore the `Task.Run` should remain.

Reply:
My main concern here isn't threads waiting in this case, just making sure that the potentially long running operation of Diffing isn't being executed on either the UI thread or the DB thread. 

Since the operation could take a second or more and it isn't itself async (so not many opportunities to handle other tasks), whatever thread ends up executing this could stay busy for a second or more, which we want to avoid on the UI or DB thread.